### PR TITLE
feat: 🎸 use request pool to track message sending

### PR DESCRIPTION
**Pull request by hcfw007**

### CODE REVIEW

File: src/puppet-mixins/message.ts [hunk 1]
```diff
@@ -187,6 +188,9 @@ export async function messageSend (this: PuppetWhatsApp, conversationId: string,
   const messageId = msg.id.id
   const cacheManager = await this.manager.getCacheManager()
   await cacheManager.setMessageRawPayload(messageId, msg)
+
+  const requestPool = RequestPool.Instance
+  await requestPool.pushRequest(messageId, 10 * 1000)
```
Comment (su-chang) [inline src/puppet-mixins/message.ts:193]: The `timeout` param should add to `options` in `messageSend()`, and set a default timeout.

File: src/puppet-whatsapp.ts [hunk 2]
```diff
@@ -181,6 +182,8 @@ class PuppetWhatsApp extends PUPPET.Puppet {
 
   private async onMessage (message: PUPPET.EventMessagePayload): Promise<void> {
     logger.info('onMessage(%s)', JSON.stringify(message))
+    const requestPool = RequestPool.Instance
```
Comment (su-chang) [inline src/puppet-whatsapp.ts:185]: I think this logic should move to `Manager` class.

File: src/request/requestPool.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { WA_ERROR_TYPE } from '../exceptions/error-type.js'
+import WAError from '../exceptions/whatsapp-error.js'
+export class RequestPool {
```
Comment (su-chang) [inline src/request/requestPool.ts:3]: Need  to add a method to clear `poolMap` when the bot is logout.

File: src/puppet-whatsapp.ts (hunk 2 above)
Comment (hcfw007) [reply src/puppet-whatsapp.ts:185]: There are multiple methods in Manager that triggers 'message' event. Any reason why we should move this?

File: src/puppet-mixins/message.ts (hunk 1 above)
Comment (hcfw007) [reply src/puppet-mixins/message.ts:193]: Good point.

File: src/request/requestPool.ts (hunk 3 above)
Comment (hcfw007) [reply src/request/requestPool.ts:3]: Indeed

File: src/puppet-whatsapp.ts (hunk 2 above)
Comment (su-chang) [reply src/puppet-whatsapp.ts:188]: The bot send message's callback valid event will listen on this method: `onMessageAck()` only.

**Review by su-chang — APPROVED** [review]

**Pull request merged**